frontend/lib/Repository/EndpointRepository/RestEndpointRepository.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:adpv_frontend/Repository/EndpointRepository/AbstractEndpointRepository.dart';
+import 'package:adpv_frontend/Repository/EndpointRepository/RestEndpointService.dart';
+import 'package:dio/dio.dart';
+
+import '../../DataModels/EndpointData.dart';
+import '../../DataModels/EndpointSummary.dart';
+
+class RestEndpointRepository implements AbstractEndpointRepository {
```

Comment:
Imo to nie jest dobre rozwiązanie mieszać repozytorium restowe z cashowym. Rozważmy przypadek, w którym chcielibyśmy pobrać nowe dane. W tym momencie to nie będzie wykonalne, ponieważ zawsze będzie pobierał dane z cash'a, jak już je raz załaduje. Myślę, że powinniśmy zrobić  dwa repozytoria, jedno `Rest`, drugie `Cash`, a jakiś komponent wyżej (może jakiś `gateway`?), powinno decydować, skąd te dane brać

Reply:
myślałem też o zrobieniu jakiegoś timeouta po którym pobierzemy nowe dane, w sumie nie mam pomysłu jak to sensowniej zrobić

Reply:
Ja bym to widział w taki sposób. Mamy dwa repozytoria `Rest` i `Cash`. Na początku odpalania aplikacji korzystamy z `Rest`, żeby ściągnąć dane, a potem zapisujemy je do `Cash`. W dalszej części decydujesz sobie, skąd chcesz te dane wziąć, czy z `Rest`, czy z `Cash` i takie repozytorium podajesz. Powinny one współdzielić jeden interfejs, dzięki czemu można by było je łatwo podmieniać. Takie rozwiązanie powinno być dużo wygodniejsze przyszłościowo. Samą idee masz dobrą, tylko trzeba to na dwie osobne klasy rozbić

Reply:
Dobra, gdzie podejmiesz decyzje z czego pobierasz? brakuje mi tylko tego w rozumowaniu